### PR TITLE
Use forked NIO with embedded Baggage in ChannelHandlerContext

### PR DESCRIPTION
**Pull request by slashmo**
The HTTP headers injecting/extracting handlers now make use of the stored Baggage inside the provided `ChannelHandlerContext`. I also added an "end-to-end" unit-test that combines both handlers in a somewhat realistic scenario.

### CODE REVIEW

**Comment by slashmo** [comment]
@ktoso Could you please take a look at this so we can merge this in (if agreed upon).

File: Package.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ let package = Package(
             url: "https://github.com/slashmo/gsoc-swift-baggage-context.git",
             from: "0.5.0"
         ),
-        .package(url: "https://github.com/apple/swift-nio.git", from: "2.17.0"),
+        .package(url: "https://github.com/slashmo/swift-nio.git", .branch("feature/baggage-context")),
```
Comment (ktoso) [inline Package.swift:17]: nice thanks

File: Sources/NIOInstrumentation/HTTPHeadersExtractingHandler.swift [hunk 2]
```diff
@@ -16,22 +16,17 @@ import Instrumentation
...
 
     public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
         guard case .head(let head) = unwrapInboundIn(data) else { return }
         var baggage = Baggage.topLevel
-        self.instrument.extract(head.headers, into: &baggage, using: HTTPHeadersExtractor())
-        self.onBaggageExtracted(baggage)
+        InstrumentationSystem.instrument.extract(head.headers, into: &baggage, using: HTTPHeadersExtractor())
+        context.baggage = baggage
+        context.fireChannelRead(data)
     }
```
Comment (ktoso) [inline Sources/NIOInstrumentation/HTTPHeadersExtractingHandler.swift:31]: nice

**Review by ktoso — APPROVED** [review]
LGTM, thanks 👍

**Pull request merged**